doc/running-simple-app.md

```diff
@@ -0,0 +1,196 @@
+directory, refer to the user-getting-started guides (for example, 
+[doc/user-getting-started-c.md](/doc/user-getting-started-c.md)).
+
+Mounting a file system
+------------
+
+To run applications in Mystikos, we need to mount the directory containing the
+application and dependencies as a file system. There are three different
+ways to mount a file system in Mystikos.
+
+### Root file system (rootfs) mount
+
+Every application will have a root file system mount. This file system will contain 
+the entry point into your application. This guide will focus mainly on mounting a
+rootfs.
+
+
+> Unless you need to mount additional directories beyond your application directory, you will only
+need a root file system mount.
+
+### Configuration Mount
+
+This mount is specified in configuration and handled by Mystikos. File system changes do not require a new application image. Refer to [doc/design/mount-config-design.md](/doc/design/mount-config-design.md) for more details.
+
+### Explicit Mount (`mount()` syscall)
+
+Mounting can also be left to the application. Refer to
+[Mounting EXT2 images from applications](#mounting-ext2-images-from-applications-(explicit-mount-or-"app-mounted"))
+for instructions.
+
+File System Types
+------------
+
+Before we can mount the application directory, we must first generate a file system image.
+The file system image generated determines which file system type is used. There are currently
+three file systems in Mystikos. 
+
+### CPIO
+
+Choose this for small applications. The entire file system will be loaded into memory when the
+application runs. When packaging your application for production, your application directory will
+be part of the executable image.
+
+### Ext2
+
+Choose this for larger applications; EXT2 uses less memory than CPIO. The file system image is a
+separate file from the executable.
+
+Mystikos supports the following features:
+- Usage of ephemeral EXT2 file systems
+- Integrity checking of EXT2 file systems ([dm-verity](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html) compatible)
+- Encryption of EXT2 file systems ([dm-crypt](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html) compatible)
+
+Modifications to EXT2 file systems are discarded when the application terminates. 
+Currently, writes are cached in trusted memory, so the number of block writes is limited by
+available memory. This limitation may be overcome later with an encrypted backing store.
+
+To support these features, Mystikos runs the following modules within the
+trusted execution environment (TEE).
+
+- EXT2 file system driver (ext2)
+- Block integrity checking device
+- Block encryption device
+- Block cache device
+
+The EXT2 image itself resides on the untrusted file system. The TEE transitions
+into the untrusted execution environment to fetch blocks. The blocks are then
+decrypted and integrity-checked within the TEE.
```

Comment:
I just realized that the root file system cannot be encrypted since no application code could run to obtain the key, so the dm-crypt feature it limited to explicit mounting.